src/lara-api.ts

```diff
@@ -18,10 +19,23 @@ export const getActivityDefinition = (activity: string): Promise<Activity> => {
   return new Promise((resolve, reject) => {
     const urlRegex = /^(https:|http:)\/\/\S+/;
     if (activity.match(urlRegex)) {
-      getActivityDefinitionFromLara(activity).then(resolve);
+      getActivityDefinitionFromLara(activity)
+      .then((resource)=>{
+        if (resource.version === 1) {
+          return convertLegacyResource(resource);
+        } else {
+          return resource;
+        }
+      })
+      .then(resolve);
     } else {
       if (sampleActivities[activity]) {
-        setTimeout(() => resolve(sampleActivities[activity]), 250);
+        if (sampleActivities[activity].version === 1) {
+          const convertedActivityResource = convertLegacyResource(sampleActivities[activity]) as Activity;
+          setTimeout(() => resolve(convertedActivityResource), 250);
```

Comment:
Do you need to do this using setTimeout? I am surprised, but maybe I don't understand something.

Reply:
Perhaps this is to make sample activities behave more like regular activities that require being downloaded? 
But I'd agree that it should be taken out, or a comment added explaining why.